tpu-provisioner/internal/controller/nodepool_garbage_collector.go

```diff
@@ -0,0 +1,94 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/ai-on-gke/tpu-provisioner/internal/cloud"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type NodePoolGarbageCollector struct {
```

Comment:
Can you add a comment with the info you included in the PR description about what this does?

e.g. `Deletes node pools that have no Nodes, are in an errored state, and where the Pod that created the node pool no longer exists (the deletion reconciler would not see these b/c there are no Node objects).`